File: Nodejs/Objects/HerokuPage.js
import {waitForElement} from '../Utils/Helper'

const elements = {
  url: 'http://the-internet.herokuapp.com/login',
  usernameId: 'username',
  passwordId: 'password',
  username: 'tomsmith',
  password: 'SuperSecretPassword!',
  formXpath: '//form[@name="login"]',
  messageXpath: '//div[@id="flash"]'
}

export default class HerokuPage {
  constructor(driver, timeout) {
    this.driver_ = driver
    this.timeout_ = timeout
  }

  async login() {
    await this.driver_
      .get(elements.url)
      .waitForElementById(elements.usernameId, this.timeout_, 2000)
      .sendKeys(elements.username)
      .waitForElementById(elements.passwordId, this.timeout_, 2000)
      .sendKeys(elements.password)
      .waitForElementByXPath(elements.formXpath, this.timeout_, 2000)
      .submit()
  }

  async getMessage() {
    return await this.driver_
      .waitForElementByXPath(elements.messageXpath, this.timeout_, 2000)
      .text()
  }
}

File: CSharp(Windows)/Objects/Web/HerokuPage.cs
using OpenQA.Selenium.Remote;
using OpenQA.Selenium;
using Appium.Objects;

namespace Appium.Models
{
    public class HerokuPage: BasePage {
        private string testUrl = "http://the-internet.herokuapp.com/login";
        private string username = "tomsmith";
        private string password = "SuperSecretPassword!";
        readonly By messageXpath = By.XPath("//div[@id='flash']");
        readonly By buttonLoginXpath = By.XPath("//button[@class='radius']");
        readonly By usernameId = By.Id("username");
        readonly By passwordId = By.Id("password");
        readonly int TIMEOUT = 15;

        public HerokuPage(RemoteWebDriver driver) : base(driver){}
        
        public void LoginWithCorrectCredentials() {
            NavigateTo(testUrl);
            driver.FindElement(usernameId).SendKeys(username);
            driver.FindElement(passwordId).SendKeys(password);
            ClickOn(buttonLoginXpath);
        }

        public string GetLoginMessage() {
            return WaitUntilVisible(messageXpath, TIMEOUT).Text;
        }
    }
}
